### ISSUE TITLE
[Feature Request] Embedded Documents

### ISSUE THREAD

**Issue by winduptoy** [balderdashy/sails-mongo#44]
I can set an attribute type to "json" to create embedded documents, but there is no way to create an enforceable schema for these embedded documents/sub-objects.  Isn't this a critical part of MongoDB schema design (embedded document vs. reference)?

**Comment by vladimiry** [comment]
+1

**Comment by leobudima** [comment]
+1

**Comment by psi-4ward** [comment]
+1

**Comment by ausir0726** [comment]
+1

**Comment by YuryT** [comment]
+1

**Comment by chrisgco** [comment]
+1

**Comment by grimor** [comment]
+1

**Comment by james075** [comment]
Hi, any updates ?

**Comment by platdesign** [comment]
+1

**Comment by mkeremguc** [comment]
+1

**Comment by tvollstaedt** [comment]
+1

**Comment by michaelpurnell** [comment]
+1

**Comment by charlieryan** [comment]
+1

**Comment by dylanbathurst** [comment]
It seems as though embedded documents are half working. When you have your associations correct in your models and you do `ModelName.create(...)` with the embedded document in the object the association and embedded document both get created. The reason it's only half working is because when you do an update on one of the documents the changes aren't reflected in embedded document or vice versa.

I was doing some testing recently and opened an issue here: https://github.com/balderdashy/waterline/issues/892

**Comment by mattapperson** [comment]
+1

**Comment by or2008** [comment]
+1

**Comment by Igor-likhomanov** [comment]
+1

**Comment by joelmarquez90** [comment]
+1

**Comment by yannickschuchmann** [comment]
+1

**Comment by particlebanana** [comment]
Ok guys this is on the feature requests list. Limiting this to collaborators only for now.

**Comment by sailsbot** [comment]
Thanks for posting, @winduptoy.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
